### DISCUSSION TITLE
Feature Request:  Drag-and-drop project files into chat.

### DISCUSSION THREAD

**Discussion by thepetshark** [cline/cline#371] [Feature Requests]
I'm happy to see the new '@' feature in chat to insert context.   Something related which I would find very useful is to just be able to drag and drop a file (or selection of files/folders)  from the VS Code 'Explorer' panel into the chat textarea.  It is usually faster and easier for me to select specific files this way than typing them out one at a time with '@path/to/module/filename'.   
This would of course open the possibility to drag and drop _anything_ from your computer into chat,  but that's a whole other discussion IMO.   Limiting it to files within the VS Code's project would be useful on its own.

Thanks for your continued work on this project,  it's a great tool!

**Comment by saoudrizwan** [comment]
I'd love to add this, but from what I've found looking into this vscode extension webviews can't handle drag and drop due to some limitation around how vscode themselves intercept drag and drop and handle it themselves. If anybody has any insight here I'd love to hear it

**Reply by ctwhome** [reply]
> If anybody has any insight here I'd love to hear it

that would be fantastic

**Comment by eljapi** [comment]
Hi guys, I'd like to have the functionality so i give it a try. What do you think? 


https://github.com/user-attachments/assets/1132e782-2894-40a7-8f7e-72ab535a285a

**Reply by ctwhome** [reply]
i just think it looks perfect!

**Comment by eljapi** [comment]
i just opened a PR https://github.com/cline/cline/pull/2676

**Reply by jralduaveuthey** [reply]
this seems like it was merged but I still cannot do that in v3.17.14
any idea why @eljapi ?
Thanks!

**Reply by eljapi** [reply]
Which OS are you using? i noticed with some friend that it works on windows draggin and holding shift into the cline chat but for others OS the built-in holding shift of vscode is not working.

**Reply by jralduaveuthey** [reply]
@eljapi Cursor in mac. with or withoug shift-holding does not work

**Comment by ianjamesburke** [comment]
Hey! Is there Currently the option to add a shortcut to add currently active file to the chat context? If not, is that something you'd be interested in adding?

**Comment by shivamklr** [comment]
Hello team,

I noticed that the current drag-and-drop functionality is limited to image files, which can be a frustrating limitation for users trying to attach documents, archives, or other assets.

To improve this core user experience, I've submitted PR #6308. It expands this functionality to allow any file type to be dropped, resolving the long-standing issue #5039. The implementation also refactors the handling logic slightly to make it easier to add file-specific validations in the future.

Feedback and reviews on the implementation would be greatly appreciated.

Best,

Shivam Kalra
[kaltq.com](https://kaltq.com)
